src/common/ui/style/index.js

```diff
@@ -2,15 +2,25 @@ import options from '../../options';
 import './style.css';
 
 let style;
+const CACHE_KEY = 'cacheCustomCSS';
+
+const setStyle = (css) => {
+  if (css && !style) {
+    style = document.createElement('style');
+    document.head.appendChild(style);
+  }
+  if (css || style) {
+    css = css || '';
+    style.textContent = css;
+    localStorage.setItem(CACHE_KEY, css);
+  }
+};
+
+setStyle(localStorage.getItem(CACHE_KEY));
```

Comment:
Why are you relying on localStorage not being [accidentally] edited by the user in devtools? Using the browser.extension.getBackgroundPage() is more reliable, especially in Firefox where just accessing `localStorage` object will throw with a security error for users who disable DOM storage in about:config.

Reply:
As [the document of getBackgroundPage](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/extension/getBackgroundPage) says:

> Note: This method cannot be used in Private Browsing mode — it always returns an empty array. For more info see bug 1329304.

> especially in Firefox where just accessing localStorage object will throw with a security error for users who disable DOM storage in about:config.

Just tested, `browser.extension.getBackgroundPage()` does not work in private browsing mode, but `localStorage` does. I'd better use `try / catch` here.

> Why are you relying on localStorage not being [accidentally] edited by the user in devtools?

If the user modifies things in devtools, everything becomes unreliable, so does `browser.extension.getBackgroundPage()`. I think this is not the case we want to cover.

> There are/were exceptions in Firefox such as private mode or containers (maybe not, I've tested it a year ago)

I may have experienced this also but I can't find the reference. Anyway, it worked just now.

Reply:
> Why are you relying on localStorage not being [accidentally] edited by the user in devtools?

Also, the cache will be updated once options are loaded from background, and every time user changes `customCSS`, so this won't be a problem.